node/forest_libp2p/src/service.rs

```diff
@@ -103,8 +125,12 @@ impl Libp2pService {
 
         let (network_sender_in, network_receiver_in) = channel(20);
         let (network_sender_out, network_receiver_out) = channel(20);
+
+        let bs_request_table = Mutex::new(HashMap::new());
```

Comment:
You don't actually need the mutex anymore since the table is just used in the libp2p service. You just need to make the self param in the function `mut` and you're good